File: elbit-consumer/src/main/java/me/antonyderham/GameSalesRecord.java
package me.antonyderham;

import com.fasterxml.jackson.annotation.JsonProperty;

public class GameSalesRecord {
    @JsonProperty("Rank")
    private double rank;

    @JsonProperty("Name")
    private String name;

    @JsonProperty("Platform")
    private String platform;

    @JsonProperty("Year")
    private String year;

    private String genre;

    @JsonProperty("Genre")
    public String getGenre() {
        return genre;
    }

    @JsonProperty("Publisher")
    private String publisher;

    @JsonProperty("NA_Sales")
    private double naSales;

    @JsonProperty("EU_Sales")
    private double euSales;

    @JsonProperty("JP_Sales")
    private double jpSales;

    @JsonProperty("Other_Sales")
    private double otherSales;

    private double globalSales;

    @JsonProperty("Global_Sales")
    public double getGlobalSales() {
        return globalSales;
    }
}

File: elbit/src/main/kotlin/me/antonyderham/GameSalesRecord.kt
package me.antonyderham

import com.fasterxml.jackson.annotation.JsonProperty

class GameSalesRecord {
    @JsonProperty("Rank")
    private val rank = 0.0

    @JsonProperty("Name")
    private val name: String? = null

    @JsonProperty("Platform")
    private val platform: String? = null

    @JsonProperty("Year")
    private val year: String? = null

    @get:JsonProperty("Genre")
    val genre: String? = null

    @JsonProperty("Publisher")
    private val publisher: String? = null

    @JsonProperty("NA_Sales")
    private val naSales = 0.0

    @JsonProperty("EU_Sales")
    private val euSales = 0.0

    @JsonProperty("JP_Sales")
    private val jpSales = 0.0

    @JsonProperty("Other_Sales")
    private val otherSales = 0.0

    @get:JsonProperty("Global_Sales")
    val globalSales = 0.0
}
